tests/phpt/regexp/009_preg_match_flags.php

```diff
@@ -0,0 +1,95 @@
+@ok
+<?php
+
+function runPregMatchTests() {
+    // Test 1: Simple match with no flags
+    $pattern = '/foo/';
+    $subject = 'foobar';
+    preg_match($pattern, $subject, $matches);
+    echo "Test 1: ";
+    var_dump($matches);
+
+    // Test 2: Case-insensitive match
+    $pattern = '/foo/i';
+    $subject = 'FOObar';
+    preg_match($pattern, $subject, $matches);
+    echo "Test 2: ";
+    var_dump($matches);
+
+    // Test 3: Named groups
+    $pattern = '/(?<first>foo)(?<second>bar)/';
+    $subject = 'foobar';
+    preg_match($pattern, $subject, $matches);
+    echo "Test 3: ";
+    var_dump($matches);
+
+    // Test 4: UTF-8 string with Unicode flag
+    $pattern = '/\p{L}+/u'; // Matches one or more Unicode letters
+    $subject = 'Hello, 世界';
+    preg_match($pattern, $subject, $matches);
+    echo "Test 4: ";
+    var_dump($matches);
+
+    // Test 5: Empty pattern
+    $pattern = '//';
+    $subject = 'foobar';
+    preg_match($pattern, $subject, $matches);
+    echo "Test 5: ";
+    var_dump($matches);
+
+    // Test 6: Empty subject
+    $pattern = '/foo/';
+    $subject = '';
+    preg_match($pattern, $subject, $matches);
+    echo "Test 6: ";
+    var_dump($matches);
+
+    // Test 7: Pattern with optional groups and PREG_UNMATCHED_AS_NULL
+    // $pattern = '/(foo)?(bar)?/';
+    // $subject = 'foo';
+    // preg_match($pattern, $subject, $matches, PREG_UNMATCHED_AS_NULL);
+    // echo "Test 7: ";
+    // var_dump($matches);
```

Comment:
I would write comments with explanation why there are commented code blocks in tests. For now it's unclear: such tests don't pass in `k2` mode, or in `server` mode, are they going to be fixed or not, etc

Reply:
Added an explanation